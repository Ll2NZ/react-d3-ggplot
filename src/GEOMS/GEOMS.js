import React            from "react";
import PropTypes        from "prop-types";
import { ChartContext } from "../_context/";

function GEOMS(props) {
        return(
            <ChartContext.Provider value={ props }>
                <svg width={ props.dimensions.width } height={ props.dimensions.height }>
                    { props.children }
                </svg>
            </ChartContext.Provider>
        );
}

GEOMS.propTypes = {
    dimensions: PropTypes.object,
    children: PropTypes.any
};

export default GEOMS;
