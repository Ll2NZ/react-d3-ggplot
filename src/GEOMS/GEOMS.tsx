import React, { ReactChildren, ReactComponentElement, ReactChild } from "react";
import { ChartContext, IContext } from "../_context/";

interface IProps extends IContext {
  children?: ReactChild[] | ReactChild;
}

function GEOMS(props: IProps) {
  if (!props.data || !props.aes || !props.dimensions) {
    throw new Error(`GEOMS must have the following properties: data, aes, and dimension`);
  }

  if (!props.children) {
    throw new Error("GEOMS expects at least one child components such as: <Rects/> or <Points/>");
  }

  return (
    <ChartContext.Provider value={props}>
      <svg width={props.dimensions.width} height={props.dimensions.height}>
        {props.children}
      </svg>
    </ChartContext.Provider>
  );
}

export default GEOMS;
