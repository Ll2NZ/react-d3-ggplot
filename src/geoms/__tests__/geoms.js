import React from 'react';
import { render } from '@testing-library/react';
import { cleanup } from '@testing-library/react';
import GEOMS from '../geoms.js';

afterEach(cleanup);

test('Underlying SVG markup should not change when we change our component logic', () => {
  // changing validState will change the markup
  const validState = {
    data: [{ x: 0, y: 0 }, { x: 10, y: 10 }],
    aes: ['x', 'y'],
    dimensions: { width: 500, height: 300, padding: 50 }
  };

  // should only render an <svg/> element
  const { container } = render(<GEOMS {...validState} />);

  expect(container.firstChild).toMatchSnapshot();
});
