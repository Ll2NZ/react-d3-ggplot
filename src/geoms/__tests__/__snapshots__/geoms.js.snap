// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Underlying SVG markup should not change when we change our component logic 1`] = `
<svg
  height="300"
  width="500"
/>
`;
